finality-aleph/src/sync/mock/backend.rs

```diff
@@ -352,3 +361,42 @@ impl ChainStatus<MockBlock, MockJustification> for Backend {
         }
     }
 }
+
+#[derive(Debug)]
+pub enum VerifierError {
+    IncorrectJustification,
+    IncorrectSession,
+}
+
+impl Display for VerifierError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), FmtError> {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl Verifier<MockJustification> for Backend {
+    type Error = VerifierError;
+
+    fn verify(
+        &mut self,
+        justification: MockJustification,
+    ) -> Result<MockJustification, Self::Error> {
+        let top_number = self.top_finalized().unwrap().header.id.number;
+        let storage = self.inner.lock();
+        let current_session = storage
+            .session_boundary_info
+            .session_id_from_block_num(top_number);
+        let justification_session = storage
+            .session_boundary_info
+            .session_id_from_block_num(justification.id().number);
+        if !(justification_session == current_session
```

Comment:
Personal preference: can we apply De Morgan's here :stuck_out_tongue:?  Would be more immediate to understand this condition imho.

Reply:
done